packages/jsii-pacmak/lib/targets/go/types/class.ts

```diff
@@ -163,3 +156,36 @@ export class ClassMethod implements GoTypeMember {
     );
   }
 }
+
+export class StaticMethod extends ClassMethod {
+  public constructor(
+    public readonly parent: GoClass,
+    public readonly method: Method,
+  ) {
+    super(parent, method);
+  }
+
+  public emit({ code }: EmitContext) {
+    const name = `${this.parent.name}_${this.name}`;
+
+    const parameters = parametersToString(this.parent, this.method.parameters);
+    code.openBlock(`func ${name}(${parameters}) ${this.returnType}`);
+
+    this.runtimeCall.emit(code);
+
+    code.closeBlock();
+    code.line();
+  }
+}
+
+function parametersToString(parent: GoClass, params: Parameter[]): string {
+  const parameters = params.map((p) => {
+    const paramName = substituteReservedWords(p.name);
+    const paramType = new GoTypeRef(parent.pkg.root, p.type).scopedName(
+      parent.pkg,
+    );
+    return `${paramName} ${paramType}`;
+  });
+
+  return parameters.length === 0 ? '' : parameters.join(', ');
+}
```

Comment:
Can this just be defined as a dynamic getter on `ClassMethod` so its inherited by `StaticMethod` and not free floating?

```suggestion
private get parametersString(): string {
  const parameters = this.method.parameters.map((p) => {
    const paramName = substituteReservedWords(p.name);
    const paramType = new GoTypeRef(this.parent.pkg.root, p.type).scopedName(
      this.parent.pkg,
    );
    return `${paramName} ${paramType}`;
  });

  return parameters.length === 0 ? '' : parameters.join(', ');
}
```